feat: Added Dashboard to price section

chetanr25 opened the pull request:
### What
Created a user dashboard taking inspiration from https://prices.openfoodfacts.org/dashboard

### Changes
- Added Dashboard ListTile option under prices section [packages/smooth_app/lib/pages/preferences/user_preferences_prices.dart](https://github.com/openfoodfacts/smooth-app/compare/develop...chetanr25:smooth-app:draft1_dashboard?expand=1#diff-904beaf5e47ade3c0dd4752fe02059a5bf1a3b947fe8a27b8d7945532ae7fb85)
- Created a UserProfile model which stores information returned by [/api/v1/users/$username](https://prices.openfoodfacts.org/api/v1/users/chetanr25?app_name=Open+Prices+Web+App) which is used in first section of dashboard. [packages/smooth_app/lib/data_models/users_profile_data.dart](https://github.com/openfoodfacts/smooth-app/compare/develop...chetanr25:smooth-app:draft1_dashboard?expand=1#diff-b4e38d172a5f7bbe74a7598ce65a3e1200c521cd5322c0fa41c1e9ef6f96e14f])
- First section of the dashboard includes price_count, product_count, proof_count, location_count etc. [ + packages/smooth_app/lib/pages/prices/prices_user_profile.dart](https://github.com/openfoodfacts/smooth-app/compare/develop...chetanr25:smooth-app:draft1_dashboard?expand=1#diff-30ae250e1b28608bec5f97fb7ddddf904d44c09e83b1e611a18baebcfe2f42c3)
- Main dashboard is built in [ + packages/smooth_app/lib/pages/prices/prices_dashboard.dart](https://github.com/openfoodfacts/smooth-app/compare/develop...chetanr25:smooth-app:draft1_dashboard?expand=1#diff-ca76380820152b2a0623de75d81761f5bf73058fdfaf46e585f0ab16946657cb)

- Prices and Proof button in dashboard currently onClicking will navigate to existing "My prices" and "My proofs" screen. (note: The list of prices or proofs isn't effected by the category selected "My Consumption" / "Other Contributions")
- The toggle button "My Consumption" and "Other Contributions" when toggled will update price and proof contributions count based on the kind of contribution. 
- The recent contributions summary which is the last section of dashboard. We will display recent contributions of the user filtered by the kind of contribution ("Community" or "Consumption") but for this we need to add `kind` parameter in GetPricesParameters() of OpenFoodFacts package in [lib/src/prices/get_prices_parameters.dart](https://github.com/openfoodfacts/openfoodfacts-dart/blob/master/lib/src/prices/get_prices_parameters.dart). I have made a PR for this in openfoodfacts-dart repo https://github.com/openfoodfacts/openfoodfacts-dart/pull/1064

### Screenshot

https://github.com/user-attachments/assets/6818d97b-3998-4f01-8b9e-bf024c96247f
<img width="311" alt="image" src="https://github.com/user-attachments/assets/e5c6037c-6eb2-433e-94a2-375f94469ca3" />



### Fixes bug(s)
- Closes:  #5206 

### Part of 
- #5206 

I’ve tried to stick to the existing codebase conventions for this. If I’ve missed something or anything needs to be changed, please let me know, I will make the updates.

Commit:
Built dashboard

Commit:
fix: build errors

chetanr25 commented:
I had made changes to the Flutter package in my local system to test if this works. I have added the kind parameter in my PR to extend `GetPricesParameters()` for filtering contributions by type (“Community” or “Consumption”). Since the kind parameter isn’t available yet in the OpenFoodFacts package and I’m trying to use it in this PR, the build test is failing.

Should I comment it out or remove it for now?

chetanr25 commented:
Once the changes from https://github.com/openfoodfacts/openfoodfacts-dart/pull/1065 are available in our Flutter package, we can delete this file [packages/smooth_app/lib/data_models/users_profile_data.dart](https://github.com/openfoodfacts/smooth-app/pull/6559/files#diff-b4e38d172a5f7bbe74a7598ce65a3e1200c521cd5322c0fa41c1e9ef6f96e14f) and use the PriceUser model instead.

Untill then should the PR be a Draft PR or is it okay to keep it open?

monsieurtanuki commented on packages/smooth_app/lib/pages/preferences/user_preferences_prices.dart:
```diff
@@ -45,6 +46,15 @@ class UserPreferencesPrices extends AbstractUserPreferences {
     return <UserPreferencesItem>[
+      if (isConnected)
+        _getListTile(
+            'My Dashboard',
```
To be localized in `app_en.arb`.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
Please rename as `PricesDashboardPage` (and the file too).

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+  Widget build(BuildContext context) {
+    return SmoothScaffold(
+      appBar: SmoothAppBar(
+        title: const Text('My Dashboard'),
```
Localize

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+        title: const Text('My Dashboard'),
+        actions: <Widget>[
+          IconButton(
+            tooltip: 'Open Prices Dashboard in browser',
```
Localize

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+class PricesDashboard extends StatelessWidget {
+  PricesDashboard({super.key});
```
```suggestion
  PricesDashboard();
```

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+            if (snapshot.hasError) {
+              return Text(snapshot.error!.toString());
+            }
+            final UserProfile userProfile = snapshot.data!.value;
```
Should check `snapshot.connectionState != ConnectionState.done`

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,202 @@
+
+class PricesDashboardWidget extends StatefulWidget {
+  const PricesDashboardWidget({super.key, required this.userProfile});
+  final UserProfile? userProfile;
```
```suggestion
  final UserProfile userProfile;
```

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,202 @@
+}
+
+class _PricesDashboardWidgetState extends State<PricesDashboardWidget> {
+  int selectedIndex = 0;
```
Would be faster to make it a `String` directly.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_user_profile.dart:
```diff
@@ -0,0 +1,92 @@
+        shrinkWrap: true,
+        children: <Widget>[
+          ListTile(
+            leading: const Icon(Icons.person, size: DEFAULT_ICON_SIZE * 2),
```
Forget about the sizes and colors.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_user_profile.dart:
```diff
@@ -0,0 +1,92 @@
+    );
+  }
+
+  Widget profileStatsButton(
```
```suggestion
  Widget _profileStatsButton(
```

teolemon commented:
@ttalex @raphodn I definitely would put My Receipts here. The wording is a bit creepy on mobile, given they are public (we do mention they will be public on upload, but still)

teolemon commented:
@chetanr25 can you rename "My Consumption" to "Receipts & GDPR requests", and "Other contributions" to "Price labels" ?

teolemon commented:
error • The setter 'kind' isn't defined for the type 'GetPricesParameters' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:73:11 • undefined_setter

teolemon commented:
/gemini review

Commit:
fixed new kind parameter using ContributionKind, replaced my contributions and other contributions

Commit:
Replaced title of category as suggested

chetanr25 commented:
> error • The setter 'kind' isn't defined for the type 'GetPricesParameters' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:73:11 • undefined_setter

This has been fixed. Since https://github.com/openfoodfacts/openfoodfacts-dart/pull/1064 is merged and is available in [openfoodfacts flutter package](https://pub.dev/packages/openfoodfacts), we can now use the ContributionKind enum.




> @chetanr25 can you rename "My Consumption" to "Receipts & GDPR requests", and "Other contributions" to "Price labels" ?

I have made the requested changes and committed them to the current PR. Initially, I had replicated the same titles (‘My Consumption’ and ‘Other Contributions’) from the [dashboard web version](https://prices.openfoodfacts.org/dashboard). Since we have updated the titles, is there anything else I should update in the PR to stay consistent?

Commit:
merged latest changes

monsieurtanuki commented:
> > error • The setter 'kind' isn't defined for the type 'GetPricesParameters' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:73:11 • undefined_setter
> 
> This has been fixed. Since [openfoodfacts/openfoodfacts-dart#1064](https://github.com/openfoodfacts/openfoodfacts-dart/pull/1064) is merged and is available in [openfoodfacts flutter package](https://pub.dev/packages/openfoodfacts), we can now use the ContributionKind enum.

@chetanr25 No it hasn't been fixed, as we haven't _released_ a new version of openfoodfacts-dart yet.
I don't want to sound old-fashioned or patronizing, but somehow you need to test your code - at least run it.

chetanr25 commented:
I had tested well and only then committing to this PR.
To verify the functionality, I first made changes locally in the openfoodfacts Flutter package within the .pub-cache to test the integration with smooth-app. Since Flutter packages are open source, I directly edited the package code for testing purpose. After confirming that the changes worked correctly, I contributed them to the openfoodfacts-dart repository.

The changes were made for testing purposes, to ensure that everything worked correctly on smooth-app. As a result, it appeared to work without any issues on my local system.

However, I realize now that I forgot the fact that the actual package update had not yet been officially released. I mistakenly assumed that it had already been published and updated the PR.

I completely understand that directly modifying installed packages is not a good practice. I had only done it temporarily for testing. After reinstalling the package, I realized the release hadn’t happened yet.

Apologies for the confusion.

monsieurtanuki commented:
> @TTalex @raphodn I definitely would put My Receipts here. The wording is a bit creepy on mobile, given they are public (we do mention they will be public on upload, but still)

For the record I still haven't understood which prices are placed where.

TTalex commented:
> For the record I still haven't understood which prices are placed where.

Yeah, it's not obvious :/

By default :

- Consumption = Receipts & GDPR
- Community = Price tag & Shop import

Receipts can be added to "Community" if user opt-outs, using boolean `owner_consumption` in `/api/v1/proofs/upload` API (useful if you add receipt that are not yours)

monsieurtanuki commented:
> > For the record I still haven't understood which prices are placed where.
> 
> Yeah, it's not obvious :/
> 
> By default :
> 
> * Consumption = Receipts & GDPR
> * Community = Price tag & Shop import

Thank you @TTalex for your explanations!
My point was also: how relevant is that distinction for a simple user? Wouldn't we be better off displaying simply 4 sections - Receipts, GDPR, Price tag and Shop import?
You would still keep your consumption/community flag.

TTalex commented:
> Thank you @TTalex for your explanations! My point was also: how relevant is that distinction for a simple user? Wouldn't we be better off displaying simply 4 sections - Receipts, GDPR, Price tag and Shop import? You would still keep your consumption/community flag.

For now, I would say that the distinction isn't really relevant for most users, and sometimes even causes confusion.
But it's a first step towards having info that is relevant to your consumption habits. "How much did I pay for my eggs last time", "What was the brand of that amazing salmon I ate two weeks ago". And more complex stats like "What part of my purchases include organic products".
In theory, the first two questions can already be answered with the current distinction, but would be very hard without, especially for active contributors (for example, ~3% of my contributed prices are products I bought).

Now, in regard to having 4 sections. GDPR is barely used (few users have done it) and shop import shouldn't land in user profiles (specific-shop profiles are used if I'm not mistaken). So in most cases, you end up with only two sections filled : Receipts and Price tags.
If you add the caveat that receipts can be declared as either owned by the user or not, you end up with a division that isn't really helpful to the user, or at least doesn't help with answering the original questions quoted above.

Anyway ... work in progress I guess 😅, thanks for raising a great question !

_(sorry for the wall of text that doesn't really help this PR)_

monsieurtanuki commented on packages/smooth_app/lib/data_models/users_profile_data.dart:
Irrelevant now as you know well: we should use `PriceUser` from the latest openfoodfacts-dart package.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+  }
+
+  // TODO(chetanr25): To be implemented in OpenFoodFacts flutter package
+  static Future<MaybeError<UserProfile>> _fetchUserProfile() async {
```
Is now implemented in openfoodfacts-dart.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard.dart:
```diff
@@ -0,0 +1,78 @@
+    );
+
+    final http.Response response =
+        await HttpHelper().doGetRequest(uri, uriHelper: uriHelperFoodProd);
```
No, not PROD, use `ProductQuery.uriPricesHelper`.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_user_profile.dart:
```diff
@@ -0,0 +1,92 @@
+              ],
+            ),
+            Text(label,
+                style: const TextStyle(
+                  fontSize: 15,
+                  fontWeight: FontWeight.bold,
```
Forget about the sizes, styles and colors.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_user_profile.dart:
```diff
@@ -0,0 +1,92 @@
+    return SmoothCard(
+      color: Theme.of(context).colorScheme.onSurface.withAlpha(24),
+      child: Container(
+        width: MediaQuery.sizeOf(context).width / 2 - 3 * LARGE_SPACE,
```
Please drop that.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,201 @@
+  }
+
+  /// Toggle between "My Consumption" and "Other Contributions"
+  Widget categorySwitch() {
```
```suggestion
  Widget _categorySwitch() {
```

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,201 @@
+              child: customToggleButton(
+                  0,
+                  Icons.shopping_cart,
+                  'Receipts & GDPR requests',
```
Localize.

monsieurtanuki commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,201 @@
+    );
+  }
+
+  Widget priceProofButton(
```
```suggestion
  Widget _priceProofButton(
```

monsieurtanuki reviewed (commented):
Hi @chetanr25!
I had pending comments that I forgot to submit, sorry about that.
Now that we have the latest openfoodfacts-dart release some parts of your code can be removed or simplified.

teolemon committed:
Merge branch 'develop' into draft1_dashboard

Commit:
Merge remote-tracking branch 'upstream/develop' into draft1_dashboard

Commit:
Merge: accepting all incoming changes

Commit:
Merge branch 'draft1_dashboard' of https://github.com/chetanr25/smooth-app into draft1_dashboard

Commit:
made profileStatsButton function to private as suggested, replaced temporvary UserProfile model with PriceUser model which is created in openfoodfacts flutter package

Commit:
Deleted un-necessary temporary model UserProfile, as we have PriceUser in dart package

teolemon commented:
error • Target of URI doesn't exist: 'package:smooth_app/data_models/users_profile_data.dart' • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:4:8 • uri_does_not_exist
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:15:32 • non_type_as_type_argument
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:35:38 • non_type_as_type_argument
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:38:40 • non_type_as_type_argument
  error • Undefined class 'UserProfile' • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:45:19 • undefined_class
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:58:28 • non_type_as_type_argument
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:69:27 • non_type_as_type_argument
  error • Undefined name 'UserProfile' • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:70:[11](https://github.com/openfoodfacts/smooth-app/actions/runs/15579009157/job/43869651613#step:8:12) • undefined_identifier
  error • The name 'UserProfile' isn't a type, so it can't be used as a type argument • packages/smooth_app/lib/pages/prices/prices_dashboard.dart:76:23 • non_type_as_type_argument
  error • Target of URI doesn't exist: 'package:smooth_app/data_models/users_profile_data.dart' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:5:8 • uri_does_not_exist
  error • Undefined class 'UserProfile' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:16:9 • undefined_class
   info • Unnecessary use of a null check ('!') • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:37:44 • unnecessary_null_checks
  error • Undefined class 'UserProfile' • packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:[15](https://github.com/openfoodfacts/smooth-app/actions/runs/15579009157/job/43869651613#step:8:16)5:7 • undefined_class

Commit:
Localized

Commit:
Merging develop to this branch

Commit:
deleted UserProfile and replaced it with PriceUser

chetanr25 commented:
I was actually waiting for https://github.com/openfoodfacts/openfoodfacts-dart/pull/1085 to be released in the openfoodfacts Flutter package so I could use it here. 

I'll remove this https://github.com/openfoodfacts/smooth-app/blob/6eea995349501dfbdd131aa146929b40bbe8eca7/packages/smooth_app/lib/pages/prices/prices_dashboard_page.dart#L61 once the package is being published.

monsieurtanuki commented:
@chetanr25 I've just released openfoodfacts 3.23.0.

Commit:
Remove _fetchUserProfile which is no longer used, added a placeholder to TitleElementType.PERCENTAGE in knowledge_pannel_builder.dart

Commit:
Changed openfoodfact package version

chetanr25 commented:
I have updated the package version in `pubspec.yaml` file. 

For now, i have added a TODO for `TitleElementType.PERCENTAGE` as this was introduced to openfoodfacts flutter package in PR https://github.com/openfoodfacts/openfoodfacts-dart/pull/1086 was causing an error in `packages/smooth_app/lib/knowledge_panel/knowledge_panels_builder.dart` file. 

https://github.com/openfoodfacts/smooth-app/blob/933bda2c528850f08ea8b5c05d9a5935671375d7/packages/smooth_app/lib/knowledge_panel/knowledge_panels_builder.dart#L333

I haven't mentioned username in TODO

monsieurtanuki commented:
> I haven't mentioned username in TODO

@chetanr25 Oh you have to. I suggest @PrimaelQuemerais.

Commit:
Added username in Knowledge panel todo

Commit:
Scrolling issue fixed, minor UI improvements

chetanr25 commented:
In the previous commit https://github.com/openfoodfacts/smooth-app/pull/6559/commits/7075c4d46a3632dae85c1f075b0184e1397541a3 , I resolved the scrolling functionality issue by refactoring the implementation. 

Previously, I was directly using `ProductPricesList()` within `PriceDashboardWidget()`, which caused scrolling behavior problems as `ProductPricesList()` was using `InfiniteScrollList` and I couldn't find a way to change scroll physics of this widget so that the scroll in `PriceDashboardWidget()` would look natural. 

After analyzing the codebase, I took inspiration from `ProductPricesList`'s implementation and used `PriceProductWidget` and `PriceDataWidget` components directly.

https://github.com/openfoodfacts/smooth-app/blob/7075c4d46a3632dae85c1f075b0184e1397541a3/packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart#L82

This solution resolves the scrolling issues, I'd particularly appreciate review on:
- Whether this implementation maintains good widget reusability or is it not the correct way to implement as if we change anything in `ProductPricesList()`, we have to do the same change in this dashboard widget.
- If there might be a better way to achieve the same functionality


[Before](https://github.com/user-attachments/assets/6c02ada2-5825-44ad-9815-04b1f06cd2f8) / [After](https://github.com/user-attachments/assets/679f60ca-b82e-4955-bf48-28539b72f88b)

{Edited because the videos weren't visible]

monsieurtanuki commented:
@chetanr25 I think you'd be better off using `InfiniteScrollList` given that:
* it'd be easier to maintain
* sticking 100% to the current web UI/UX isn't even required, as @teolemon stated [here](https://github.com/openfoodfacts/smooth-app/pull/6589#issuecomment-2868726417)

This PR code could be very simple, with a very simple UI/UX.
I don't know what they'll decide about the UI/UX.

chetanr25 commented:
> @chetanr25 I think you'd be better off using `InfiniteScrollList` given that:
> 
> * it'd be easier to maintain
> * sticking 100% to the current web UI/UX isn't even required, as @teolemon stated [here](https://github.com/openfoodfacts/smooth-app/pull/6589#issuecomment-2868726417)
> 
> This PR code could be very simple, with a very simple UI/UX. I don't know what they'll decide about the UI/UX.

I tried to use `InfiniteScrollList` but this didn't work well in this case since the dashboard uses a variety of widgets and not `List<T>`

> I think you'd be better off using `InfiniteScrollList` given that:

Just to clarify, do you suggest that we go back to the earlier approach (in before video)?
  
I implemented this (similar to [product_prices_list.dart](https://github.com/openfoodfacts/smooth-app/blob/develop/packages/smooth_app/lib/pages/prices/product_prices_list.dart) but without InfiniteScrollList) after keeping maintainability and re-usability in mind. 
https://github.com/openfoodfacts/smooth-app/blob/7075c4d46a3632dae85c1f075b0184e1397541a3/packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart#L82C1-L89C25

I am actually confused, how to go about this.

monsieurtanuki commented:
>  this didn't work well in this case since the dashboard uses a variety of widgets

@chetanr25 That's the point. Either you try to stick to the web UI, or you try to display and code things simply.
Personally I have no interest in that web UI that I find hard to understand.

chetanr25 committed:
Merge remote branch 'upstream/develop' into draft1_dashboard

chetanr25 committed:
feat(prices): update dashboard tile to new screen. Reverting readme to latest version in develop

chetanr25 committed:
Merge branch 'develop' into draft1_dashboard

chetanr25 committed:
fix: resolve arb file issues from merge

chetanr25 commented:
Apologies for the delay, I'm back and have made some updates.
I noticed the preferences page was redesigned in PR #6526, so I've updated the dashboard tile placement accordingly. 
In current PR, It's now accessible from:
- Community tab → "See all stats"
- "My prices" section
- "My Dashboard


https://github.com/user-attachments/assets/599216c4-cbe2-4594-844e-f696cbcdcd84


Do you have any suggestions for the dashboard placement or the overall approach?
Also, just checking if is this feature still relevant, or should I hold off?

chetanr25 commented:
The CI is failing on `flutter analyze --fatal-infos --fatal-warnings .`  due to import issues in files that are outside the scope of this PR.
Do you recommend creating a separate issue and opening another PR to fix these problems, since they are causing the CI to fail?

chetanr25 committed:
refactor: remove hardcoded width calculation in prices user profile, enhanced UI for Dashboard tile

chetanr25 commented:
### Updated UI for Prices User Profile

Removed hardcoded `MediaQuery.sizeOf(context).width` calculation and updated the profile stats to use:
- `Row` + `Expanded` pattern for responsive 2x2 grid
- Themed colors (`primaryMedium`/`primaryDark`) matching app design
- Consistent border radius and spacing

![Updated UI](https://github.com/user-attachments/assets/66757b05-5cb7-4739-b851-b26f03daf040)

monsieurtanuki commented:
Hi @chetanr25!
Your PR doesn't pass the checks.
Besides, UI isn't in my jurisdiction. For that matter you need a review from @teolemon, @g123k or @PrimaelQuemerais.

chetanr25 committed:
Merge remote-tracking branch 'upstream/develop' into draft1_dashboard

chetanr25 committed:
fix: correct named parameter order to fix flutter analyze error

chetanr25 commented:
> Hi @chetanr25! Your PR doesn't pass the checks. Besides, UI isn't in my jurisdiction. For that matter you need a review from @teolemon, @g123k or @PrimaelQuemerais.

Thanks for pointing that out.

I’ve now rebased onto the latest upstream/develop and fixed the remaining `flutter analyze --fatal-infos --fatal-warnings` issues. All CI checks are passing on the latest commits.

@teolemon @g123k @PrimaelQuemerais I'd also love your thoughts on whether the current dashboard tile button placement is appropriate, or if you would suggest a different navigation entry point for this screen.

chetanr25 commented:
One more question: the generated l10n/app_localizations*.dart files are currently being tracked should these be committed as part of this PR, or would you prefer them to be excluded?

g123k commented:
> One more question: the generated l10n/app_localizations*.dart files are currently being tracked should these be committed as part of this PR, or would you prefer them to be excluded?

They are not tracked anymore

chetanr25 committed:
Merge branch 'openfoodfacts:develop' into draft1_dashboard

chetanr25 committed:
fix(prices): Add missing VIEW_PRODUCT case handler in prices dashboard, pushing app_localizations.dart

chetanr25 commented:
Fixes analyzer error by adding missing `VIEW_PRODUCT` case handler

This PR adds the missing `ProductPriceAction.VIEW_PRODUCT` case handler in `prices_dashboard_widget.dart` to match the implementation that was added in `product_prices_list.dart` (PR #7362). 

The switch statement was non-exhaustive, causing Flutter analyzer to fail with `--fatal-warnings`. 

Also includes localization updates in `app_localizations.dart` (Which wasn't pushed before)

chetanr25 committed:
Merge branch 'develop' into draft1_dashboard

chetanr25 committed:
Merge branch 'develop' of github.com:openfoodfacts/smooth-app into draft1_dashboard

chetanr25 committed:
Merge branch 'draft1_dashboard' of github.com:chetanr25/smooth-app into draft1_dashboard

chetanr25 committed:
Implemented copiolot's suggestion. Localized values, error handling for result from OpenPricesAPIClient.getPrices, ProductPriceAction menu updated - added open_product

chetanr25 commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,343 @@
+    }
+
+    switch (res) {
+      case ProductPriceAction.VIEW_PRODUCT:
+        AppNavigator.of(
+          context,
+        ).push(AppRoutes.PRODUCT_LOADER(price.product!.code));
```
Added view_product to the menu instead of removing its functionality (in switch statement)

chetanr25 commented on packages/smooth_app/lib/pages/prices/prices_dashboard_widget.dart:
```diff
@@ -0,0 +1,343 @@
+              },
+              subtitle: Text(appLocalizations.prices_generic_title),
+              title: Text(
+                selectedCategory == 'consumption'
+                    ? profile.priceKindConsumptionCount.toString()
+                    : profile.priceKindCommunityCount.toString(),
+              ),
+              trailing: const Icon(Icons.arrow_forward),
+            ),
+          ),
+        ),
+        Expanded(
+          flex: 1,
+          child: SmoothCard(
+            child: ListTile(
+              onTap: () {
+                Navigator.of(context).push(
+                  MaterialPageRoute<void>(
+                    builder: (BuildContext context) =>
+                        const PricesProofsPage(selectProof: false),
+                  ),
+                );
+              },
+              subtitle: Text(appLocalizations.prices_proof_subtitle),
+              title: Text(
+                selectedCategory == 'consumption'
+                    ? profile.proofKindConsumptionCount.toString()
+                    : profile.proofKindCommunityCount.toString(),
```
We can probably assume `profile.priceKindConsumptionCount.toString()` and `profile.priceKindCommunityCount.toString()` will never be null

chetanr25 committed:
Merge branch 'develop' into draft1_dashboard

chetanr25 committed:
Merge branch 'develop' into draft1_dashboard
